_posts/2017-01-17-Oredev2016.md

```diff
@@ -0,0 +1,695 @@
+
+### Namespaces
+Isolated world in Kubernetes where you can have different environments inside the same cluster.
+If you want to run a TEST, DEV and PRD environment it's possible with Kubernetes to keep them isolated from each other.
+
+<p style="text-align: center;">
+   <img style="max-width:640px;" alt="Namespaces" src="/img/oredev/namespaces.png">
+ </p>
+
+### Kubernetes in Production
+
+#### HTTP load balancing
+Kubernetes does not have support for these in production so be sure to have a look at it when deploying to PRD.
+* Expose Kubernetes services to the outside world
+* SSL offloading
+* Gzip
+* Redirects - redirect every HTTP call to HTTPS
+
+#### NodePort
+* Expose service ports as ports on physical nodes
+* Beware of port conflicts!
+
+`kubectl expose deployment my-nginx --type=NodePort`
+
+#### Kubernetes Ingress
+* Level 7 load balancer configuration
+* SSL offloading
+* Support for GCE load balancers
+* Future support for extensions (not quite there yet)
+* At the moment useful on Google Cloud Engine
+
+#### Using a custom load balancer
+* Use Ha-proxy in front of Kubernetes
+* Configure Ha-proxy dynamically
+* The same works for nginx, apache
+* Ha-proxy does SSL offloading
+
+##### How does ha-proxy know about our services?
+* Ha-proxy uses a static config file
+* Auto-generate it based on data in etcd
+* Confd
+
+### Blue-green deployment
+* Deployment without downtime
+* Only one version is active at a time
+* Rolls back on failed deployment
+
+>When updating our cluster we have two options:
+We want to update our cluster with a rolling update,
+which updates one pod at a time, rather than taking down
+the entire service at the same time or you can do a Blue-green deployment where you
+make an entirely new cluster of instances and let the load balancer go from blue to green.
+
+
+<a name="ux" />
+
+# A developer's guide to the UX galaxy - Tess Ferrandez
+
+<img class="float-image p-image" src="{{ '/img/oredev/tessferrandez.jpg' | prepend: site.baseurl }}" alt="Tess Ferrandez" style="width: 100px"/>
+
+[Tess Ferrandez](https://www.twitter.com/TessFerrandez) is a full stack developer on the Microsoft Platform.
+She is equally happy debugging nasty back-end issues as she is developing apps or working on UI/UX design.
+She has been blogging at [her blogsite](http://blogs.msdn.microsoft.com/Tess) for the better part of her career, and spoken at lots and lots of conferences around the world.
+
+Software is build into two parts, it is what we see and how we do it.
+In this session Tess talks about what these features should look like.
+If you design 'an order' incorrectly and the user can't figure out to place that order, you have a bad UX design and it will cost your company a lot of money.
+Now two UX researches at Microsoft deep dived into 100 years of research of signs and symbols and listen to what people find a good interface.
```

Comment:
Now, ... (comma)

Would also rephrase the end a bit:
... and listened to what people define as a good interface.